### PR TITLE
EMBCESSMOD-1375 - Supplier Portal Error Page

### PR DESCRIPTION
**Pull request by MShayOffice**
Error page & routing for Supplier portal, redirects to error page if API or Dynamics can't be reached.
Adjustments to Dynamics check in ConfigController to make distinction between Maintenance outage and healthcheck failure.

### CODE REVIEW

File: suppliers/src/API/EMBC.Suppliers.API/ConfigurationModule/Controllers/ConfigController.cs [hunk 1]
```diff
@@ -54,16 +54,25 @@ public async Task<ActionResult<Config>> GetConfig()
...
-            var maintPage = conf["MAINTENANCE_PAGEDOWN"] ?? string.Empty;
+            var maintPageMsg = conf["MAINTENANCE_PAGEDOWN"] ?? "Default";
             var maintTimeStr = conf["MAINTENANCE_START"] ?? string.Empty;
 
             var envStr = conf["ASPNETCORE_ENVIRONMENT"] ?? string.Empty;
 
             DateTime maintTime;
 
+            // Check for CRM connectivity, immediately return error if fails
             Task<bool> healthCheck = CRMHealthcheck();
```
Comment (ytqsl) [inline suppliers/src/API/EMBC.Suppliers.API/ConfigurationModule/Controllers/ConfigController.cs:65]: did you mean `var crmHealthy = await CRMHealthcheck();'?

File: suppliers/src/API/EMBC.Suppliers.API/ConfigurationModule/Controllers/ConfigController.cs [hunk 2]
```diff
@@ -113,27 +111,48 @@ public async Task<ActionResult<Config>> GetConfig()
...
-                {
-                    siteDown = false;
-                }
-                return siteDown;
+                // If query returns list with items in it, site is healthy
+                return list.List.Count > 0;
             }
             catch (Exception e)
             {
                 Console.WriteLine(e);
```
Comment (ytqsl) [inline suppliers/src/API/EMBC.Suppliers.API/ConfigurationModule/Controllers/ConfigController.cs:127]: we have a logging framework worth using

**Review by ytqsl — APPROVED** [review]
Some nitpicks

File: suppliers/src/API/EMBC.Suppliers.API/ConfigurationModule/Controllers/ConfigController.cs (hunk 2 above)
Comment (MShayOffice) [reply suppliers/src/API/EMBC.Suppliers.API/ConfigurationModule/Controllers/ConfigController.cs:127]: @josephweinkamgov Can you look into the logging system being used, and use it if Dynamics can't be reached?
Also please look into the other adjustment Yossi recommended above. Up to you if that warrants a separate task, or if you want to reuse the original Dynamics Healthcheck one.

**Pull request merged**